dspace-api/src/test/java/org/dspace/discovery/DiscoveryIT.java

```diff
@@ -99,6 +107,9 @@ public class DiscoveryIT extends AbstractIntegrationTestWithDatabase {
     MetadataAuthorityService metadataAuthorityService = ContentAuthorityServiceFactory.getInstance()
                                                                                       .getMetadataAuthorityService();
 
+    SolrSearchCore solrSearchCore = DSpaceServicesFactory.getInstance().getServiceManager()
```

Comment:
@floriangantner : The tests are still failing here, but are no longer failing on `main`.  It now seems like there's something specific in your tests that is causing issues.

It *might* be this usage of SolrSearchCore, as we have a `MockSolrSearchCore` that is used in all tests.

To get access to the `MockSolrSearchCore`, you should be able to just do
```
@Autowired
MockSolrSearchCore solrSearchCore;
```

That said, I'm honestly not sure why this PR is still failing tests.  The failing tests *seem* unrelated to the changes you've made here, but they are tests that are also using Solr.  So, I'm not sure if this is the cause of the issues, or if there's something else that I'm missing.